content/en/blog/_posts/2024-05-13-cloud-provider-migration.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: blog
+title: 'Completing the largest migration in Kubernetes history'
+date: 2024-05-20
+slug: completing-cloud-provider-migration
+author: >
+  Andrew Sy Kim (Google),
+  Michelle Au (Google),
+  Michael McCune (Red Hat)
+---
+
+Since as early as Kubernetes v1.7, the Kubernetes project has pursued the ambitious goal of removing built-in cloud provider integrations.
+While these integrations were instrumental in Kubernetes' early development and growth, their removal was driven by two key factors:
+the growing complexity of maintaining native support for every cloud provider across millions of lines of Go code, and the desire to establish
+Kubernetes as a truly vendor-neutral platform.
+
+After many releases, we're thrilled to announce that all cloud provider integrations have been successfully migrated from the core Kubernetes repository to external plugins.
+This migration was a complex and long-running effort due to the numerous impacted components and the critical code paths that relied on the built-in integrations for the
+five initial cloud providers: Google Cloud, AWS, Azure, OpenStack, and vSphere. To successfully complete this migration, we had to build four new subsystems from the ground up:
+
+1. [Cloud Controller Manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/)
+2. [API Server Network Proxy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1281-network-proxy)
+3. [Kubelet Credential Provider Plugins](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2133-kubelet-credential-providers)
+4. [CSI In-tree Storage migration plugins](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/625-csi-migration/README.md)
```

Comment:
I'd link to KEPs consistently:
```suggestion
1. **Cloud controller manager** ([KEP-2392](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/2392-cloud-controller-manager/README.md))
1. **API server network proxy** ([KEP-1281](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1281-network-proxy))
1. **kubelet credential provider plugins ([KEP-2133](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2133-kubelet-credential-providers))
1. Storage migration to use [CSI](https://github.com/container-storage-interface/spec?tab=readme-ov-file#container-storage-interface-csi-specification-) ([KEP-625](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/625-csi-migration/README.md))
```

Feel free to put those in KEP order if you want to.